### PR TITLE
Add Buyer Questions to Event

### PR DESCRIPTION
**Pull request by ronin1972**

### CODE REVIEW

File: cat/CaT-service.yaml [hunk 1]
```diff
@@ -3296,7 +3296,32 @@ components:
...
+              items:
+                allOf:
+                  - $ref: '#/components/schemas/EvalCriteria'
+                  - type: object
+                    properties:
+                      relatesTo:
+                        type: string
+                      relateItems:
+                        type: string
+                      requirementGroups:
```
Comment (adrianmilne) [inline cat/CaT-service.yaml:3310]: Might this be better called `questionGroups`? (as RequirementGroup is a different schema in the CCS-OCDS schemas - could be a bit confusing?)

Comment (adrianmilne) [reply cat/CaT-service.yaml:3310]: Should this be an array? (multiple groups?)

File: cat/CaT-service.yaml [hunk 2]
```diff
@@ -3296,7 +3296,32 @@ components:
...
+                        allOf:
+                          - $ref: '#/components/schemas/QuestionGroup'
+                          - type: object
+                            properties:
+                              OCDS:
+                                type: object
+                                properties:
+                                  requirements:
```
Comment (adrianmilne) [inline cat/CaT-service.yaml:3318]: Similar comment as above - should this maybe be `questions` (as `requirement` is a different schema object)?

Comment (ronin1972) [reply cat/CaT-service.yaml:3318]: Unfortunately it is defined in OCDS as requirements

File: cat/CaT-service.yaml (hunk 1 above)
Comment (ronin1972) [reply cat/CaT-service.yaml:3310]: Definitely should be an array. Just tryig to work out if covered by the OCDS constraint for the name

File: cat/CaT-service.yaml (hunk 2 above)
Comment (adrianmilne) [reply cat/CaT-service.yaml]: Should it be of type `Requirement` in that case if it is OCDS? As `Question` contains `Requirement` and also non OCDS fields?

Comment (ronin1972) [reply cat/CaT-service.yaml]: Hmmm, trying to think. I think I have got myself into a bit of a mess by conflating questions (requirements) and question generation

Comment (ronin1972) [reply cat/CaT-service.yaml]: In OCDS requirement=question. The problem is we have found items outside OCDS and put them in a "nonOCDS" section at each level. This is OK for when they are separate endpoints but when you glue them all together in one structure it becomes this mess. Maybe have a chat when you are free.

File: cat/CaT-service.yaml [hunk 3]
```diff
@@ -3296,7 +3296,34 @@ components:
               type: string
               format: date-time
               description: Date/time - Event Created
-
+            buyerQuestions:
+              type: array
+              items:
+                allOf:
+                  - $ref: '#/components/schemas/EvalCriteria'
+                  - type: object
```
Comment (adrianmilne) [inline cat/CaT-service.yaml:3304]: Hi @ronin1972 - I just had a run through the generator to have a look at the java classes and this approach may not work (i.e. combining a $ref and object in this manner). The generator will just see this as an array of type `EvalCriteria` (even though it looks ok in the swagger editor).

I think we may either need to extend `EvalCriteria` to include a list of `QuestionGroups` (and likewise add a list of `Questions` to each `QuestionGroup`), or create equivalent wrapper objects to reference here.

Let me just check with @tommyb82 to see if he's aware of any workaround.

**Pull request closed without merging**